src/Control/Monad/Error/Hoist.hs

```diff
@@ -41,6 +42,12 @@ instance MonadError e' m ⇒ HoistError m (Either e) e e' where
 instance (m ~ n, MonadError e' m) ⇒ HoistError m (EitherT e n) e e' where
   hoistError f = eitherT (throwError . f) return
 
+instance MonadError e' m ⇒ HoistError m (ExceptT e Identity) e e' where
+  hoistError f = either (throwError . f) return . runExcept
+
+instance MonadError e' m ⇒ HoistError m (ExceptT e m) e e' where
```

Comment:
Honestly, I have no idea why `instance MonadError e' m ⇒ HoistError m (ExceptT e m) e e' where` doesn't result in overlapping instances. Actually, I first played around with closed type families to avoid overlapping instances, but then I found out the the version that uses the same type variable works, too.